README.md

```diff
@@ -28,6 +30,137 @@ git clone git@github.com:kamp-us/monorepo.git
+
+- Add the following text block to [your `/etc/hosts` file](https://www.howtogeek.com/27350/beginner-geek-how-to-edit-your-hosts-file/).
+
+```text
+127.0.0.1 localhost.kamp.us
+127.0.0.1 pasaport.localhost.kamp.us
+127.0.0.1 pano.localhost.kamp.us
+127.0.0.1 sozluk.localhost.kamp.us
+127.0.0.1 gql.localhost.kamp.us
+```
+
+### 3. Setting up `.env` files
+
+- Duplicate `.env.example` files and rename them as `.env` in the following folders: `db/prisma`, `apps/gql`, `apps/kampus`, `apps/pasaport`.
+
+### 4. Prisma Setup and Database Configuration
+
+- Move to `db/prisma` folder.
+
+```sh
+cd db/prisma
+```
+
+- Build/start the container. (You should have docker installed.)
+
+```sh
+docker-compose up -d
+```
+
+- Generate artifacts. (e.g. Prisma Client)
+
+```sh
+npm run prisma:generate
+```
+
+- Push the Prisma schema state to the database.
+
+```sh
+npm run prisma:push
+```
+
+### 5. Running dev servers
+
+- Run the command below at the root folder.
+
+```sh
+# Runs "gql", "kampus" and "pasaport" apps so the platform can be alive with 1 command.
+npm run web
+```
+
+- Now you can go to [localhost.kamp.us:3000](localhost.kamp.us:3000) to see it live. 🚀
+
+```sh
+# Runs all "dev" commands on each package inside the workspace
+npm run dev
+```
+
+```sh
+# Runs only "gql" server so backend work can be done separetly.
+npm run gql
+```
+
+### 6. Logging in on development server(optional)
+
+#### 6.1 GitHub
+
+- Go to [OAuth Apps page](https://github.com/settings/developers) on GitHub.
+- Click on `New OAuth App`.
+  - Application name: "you can choose any name"
+  - Homepage URL: http://pasaport.localhost.kamp.us:3001/
+  - Authorization callback URL: http://pasaport.localhost.kamp.us:3001/auth/callback/github
+- Click on `Register Application`.
+- On the application page, click on `Generate a new client secret`.
+- Make the following changes to the corresponding files.
+
+```sh
+# apps/kampus, apps/pasaport
+GITHUB_ID=<Your_Client_ID>
+GITHUB_SECRET=<Your_Client_Secret>
+```
+
+- Now you can use `Sign in with GitHub` on development server.
+
+#### 6.2 Discord
+
+- Go to [Applications page](https://discord.com/developers/applications) on Discord.
+- Click on `New Application`.
+  - Name: "you can choose any name"
+  - Click on `Create`.
+- Go to `OAuth2` on the left sidebar.
+- Click on `Reset Secret` to generate secret key.
+- Click on `Add Redirect` and enter `http://pasaport.localhost.kamp.us:3001/auth/callback/discord`
+- Save changes.
+- Make the following changes to the corresponding files.
+
+```sh
+# apps/kampus, apps/pasaport
+DISCORD_ID=<Your_Client_ID>
+DISCORD_SECRET=<Your_Client_Secret>
+```
+
+- Now you can use `Sign in with Discord` on development server.
+
+#### 6.3 Twitch
+
+- Go to [Developer Applications page](https://dev.twitch.tv/console/apps) on Twitch.
+- Click on `Register Your Application`.
+  - Name: "you can choose any name"
+  - OAuth Redirect URLs: https://pasaport.localhost.kamp.us:3001/auth/callback/twitch
```

Comment:
Twitch doesn't allow URLs with subdomains without HTTPS protocol.
![image](https://github.com/kamp-us/monorepo/assets/72770244/e6bed7fb-aa41-4972-928c-4faa6e884721)
It allows localhost URLs without HTTPS protocol.
![image](https://github.com/kamp-us/monorepo/assets/72770244/0996a93f-2812-420f-88ad-e572d76d4484)

This doesn't work right now but I wanted to add the steps regardless since I think all the other steps will remain the same.

Reply:
If you want to create development server via `ssl` (e.g: https://pasaport.localhost.kamp.us:3001/auth/callback/twitch), you may look at [local-ssl-proxy](https://www.npmjs.com/package/local-ssl-proxy) npm package. I've used that and it just works, you do not need to make anything. Maybe you also want to include to README so that every dev can use easily. Because creating SSL certificate, installing to machine then starting development server via SSL is something hard for ppl who does not know the process.

Reply:
Thanks! I've looked into it, but couldn't make it work. If someone can make it work, I'd be more than happy to arrange and include the steps required in the README.

Reply:
> Thanks! I've looked into it, but couldn't make it work. If someone can make it work, I'd be more than happy to arrange and include the steps required in the README.

See example usage:
```sh
# install concurrently to run multiple commands concurrently
npm install -g concurrently

# run 'npm run dev' and 'npx local-ssl-proxy --target 3000 --source 3001' concurrently
concurrently "npm run dev" "npx local-ssl-proxy --target 3000 --source 3001"
```

See output below:
<img width="672" alt="image" src="https://github.com/kamp-us/monorepo/assets/59085992/f650dbd7-5151-4a0c-85ed-f213edb66e1b">

Reply:
@EmirBoyaci if we want to do this pasaport, let's do it for everything. but since this is a readme update, *for now* we can skip twitch provider.

Let's create a new task for this, and @EmirBoyaci i'd love for you to lead that efforts with @sevilayerkan and @emredevsalot

Reply:
Also i think this can be done in our docker setup. cc @gurkanguray

Reply:
> @EmirBoyaci if we want to do this pasaport, let's do it for everything. but since this is a readme update, *for now* we can skip twitch provider.
> 
> 
> 
> Let's create a new task for this, and @EmirBoyaci i'd love for you to lead that efforts with @sevilayerkan and @emredevsalot 

You're right, that would make more sense. I'll be creating issue and explain the context today.

Reply:
I've commented out "Discord" and "Twitch" login steps for now until they are resolved. I think with "GitHub" login steps, the installation is sufficient for newcomers for now.

Reply:
Track issue https://github.com/kamp-us/monorepo/issues/685